shell/src/main/java/alluxio/shell/command/CheckConsistencyCommand.java

```diff
@@ -48,28 +64,87 @@ public String getCommandName() {
   public void run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     AlluxioURI root = new AlluxioURI(args[0]);
+    checkConsistency(root, cl.hasOption("r"));
+  }
+
+  private void checkConsistency(AlluxioURI path, boolean repairConsistency) throws
+      AlluxioException, IOException {
+    //load the Newly created file's metadata in alluxio
+    ListStatusOptions listOptions = ListStatusOptions.defaults();
+    listOptions.setLoadMetadataType(LoadMetadataType.Always);
+    mFileSystem.listStatus(path, listOptions);
     CheckConsistencyOptions options = CheckConsistencyOptions.defaults();
-    List<AlluxioURI> inconsistentUris = FileSystemUtils.checkConsistency(root, options);
+    List<AlluxioURI> inconsistentUris = FileSystemUtils.checkConsistency(path, options);
     if (inconsistentUris.isEmpty()) {
-      System.out.println(root + " is consistent with the under storage system.");
-    } else {
+      System.out.println(path + " is consistent with the under storage system.");
+    }
+    if (!repairConsistency) {
+      if (inconsistentUris.isEmpty()) {
+        return;
+      }
       Collections.sort(inconsistentUris);
       System.out.println("The following files are inconsistent:");
       for (AlluxioURI uri : inconsistentUris) {
         System.out.println(uri);
       }
+    } else {
+      if (inconsistentUris.isEmpty()) {
+        return;
+      }
+      Collections.sort(inconsistentUris);
+      System.out.println(path + " have: " + inconsistentUris.size() + " files inconsistent.");
+      List<AlluxioURI> dirInconsistencys = new ArrayList<AlluxioURI>();
+      for (int i = 0; i < inconsistentUris.size(); i++) {
+        URIStatus status = mFileSystem.getStatus(inconsistentUris.get(i));
+        if (status.isFolder()) {
+          dirInconsistencys.add(inconsistentUris.get(i));
+          continue;
+        }
+        System.out.println("repairing path: " + inconsistentUris.get(i));
+        DeleteOptions deleteOptions = DeleteOptions.defaults().setAlluxioOnly(true);
+        mFileSystem.delete(inconsistentUris.get(i), deleteOptions);
+        Closer closer = Closer.create();
+        try {
+          OpenFileOptions openFileOptions = OpenFileOptions.defaults()
+              .setReadType(ReadType.CACHE_PROMOTE);
+          if (mFileSystem.exists(inconsistentUris.get(i))) {
```

Comment:
When is this expected to be true, if another user creates this file between the delete and exists?